Kickstarter-iOS/Views/Controllers/LiveStreamChatViewController.swift

```diff
@@ -0,0 +1,268 @@
+
+      let vc = Storyboard.LiveStream.instantiate(LiveStreamChatViewController.self)
+      vc.delegate = delegate
+      vc.viewModel.inputs.configureWith(project: project, liveStreamEvent: liveStreamEvent, chatHidden: false)
+      vc.shareViewModel.inputs.configureWith(shareContext: .liveStream(project, liveStreamEvent))
+
+      return vc
+  }
+
+  internal override func viewDidLoad() {
+    super.viewDidLoad()
+
+    self.tableView.dataSource = self.dataSource
+    self.tableView.keyboardDismissMode = .onDrag
+    self.tableView.transform = CGAffineTransform(scaleX: 1, y: -1)
+
+    self.chatInputViewContainer.addSubview(self.liveStreamChatInputView)
+
+    NSLayoutConstraint.activate([
+      self.liveStreamChatInputView.leftAnchor.constraint(equalTo: self.chatInputViewContainer.leftAnchor),
+      self.liveStreamChatInputView.topAnchor.constraint(equalTo: self.chatInputViewContainer.topAnchor),
+      self.liveStreamChatInputView.bottomAnchor.constraint(equalTo: self.chatInputViewContainer.bottomAnchor),
+      self.liveStreamChatInputView.rightAnchor.constraint(equalTo: self.chatInputViewContainer.rightAnchor)
+      ])
+
+    self.viewModel.inputs.viewDidLoad()
+  }
+
+  internal override func bindStyles() {
+    super.bindStyles()
+
+    _ = self
+      |> baseLiveStreamControllerStyle()
+      |> UIViewController.lens.view.backgroundColor .~ .ksr_navy_700
+
+    _ = self.tableView
+      |> UITableView.lens.backgroundColor .~ .ksr_navy_700
+      |> UITableView.lens.separatorStyle .~ .none
+      |> UITableView.lens.rowHeight .~ UITableViewAutomaticDimension
+      |> UITableView.lens.estimatedRowHeight .~ 200
+
+    self.tableView.contentInset = .init(topBottom: Styles.grid(1))
+  }
+
+  //swiftlint:disable:next function_body_length
+  internal override func bindViewModel() {
+    super.bindViewModel()
+
+    NotificationCenter.default
+      .addObserver(forName: .ksr_sessionStarted, object: nil, queue: nil) { [weak self] _ in
+        self?.viewModel.inputs.userSessionStarted()
+    }
+
+    NotificationCenter.default
+      .addObserver(forName: .UIDeviceOrientationDidChange, object: nil, queue: nil) { [weak self] _ in
+        self?.viewModel.inputs.deviceOrientationDidChange(
+          orientation: UIApplication.shared.statusBarOrientation
+        )
+    }
+
+    self.viewModel.outputs.prependChatMessagesToDataSourceAndReload
+      .observeForUI()
+      .observeValues { [weak self] chatMessages, reload in
+        let indexPaths = self?.dataSource.add(chatMessages, toSection: Section.messages.rawValue)
+        indexPaths.doIfSome { self?.insert($0, andReload: reload) }
+    }
+
+    self.viewModel.outputs.openLoginToutViewController
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.openLoginTout(loginIntent: $0)
+    }
+
+    Keyboard.change
+      .observeForUI()
+      .observeValues { [weak self] change in
+        if change.notificationName == .UIKeyboardWillShow {
```

Comment:
Yeah, it works in the `CommentDialogViewController` to only observe `change.frame.minY` because it's only needed once when the view appears. Upon sending the modal is dismissed so there isn't any animation of the input field back to the bottom of the screen in that view.

All I get observing `change.frame.minY` is:
```
minY: 409.0
minY: 667.0
```

Ideally we'd want to use an input accessory view for this but it's weird when changing tabs for the input field to appear and disappear.

For now this is the best solution I could come up with.